sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1480,10 +1522,50 @@ _pi_queue::pi_queue_group_t::getZeQueue(uint32_t *QueueGroupOrdinal) {
   if (ZeResult) {
     die("[L0] getZeQueue: failed to create queue");
   }
+
+  // When using immediate commandlists, we create a commandlist corresponding
+  // to the L0 queue created above. Created once, the commandlists are reused
+  // without needing a close or reset.
+  if (UseImmediateCommandLists) {
+    ze_command_list_handle_t ZeCommandList;
+    ZE_CALL_NOCHECK(zeCommandListCreateImmediate,
+                    (Queue->Context->ZeContext, Queue->Device->ZeDevice,
+                     &ZeCommandQueueDesc, &ZeCommandList));
```

Comment:
The imm command-list is created on (context;device) and can be used on any queue. So let's cache them in the _pi_context and re-use rather than creating/destroying for every queue. This is similar to what we are doing for regular command-lists.

Reply:
Actually, I recall now that the command-list has queue's ordinal/index in the ZeCommandQueueDesc.

Reply:
So let's keep them in the queue.